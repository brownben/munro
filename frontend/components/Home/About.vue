<script setup lang="ts">
import {
  MapIcon,
  DeviceMobileIcon,
  SwitchVerticalIcon,
} from '@heroicons/vue/outline/index.js'
</script>
<template>
  <section
    class="mx-auto grid max-w-screen-lg grid-cols-1 gap-12 px-6 py-10 sm:py-16 md:grid-cols-3 md:gap-8 lg:gap-10 lg:px-8"
  >
    <div>
      <SwitchVerticalIcon
        aria-hidden="true"
        class="h-12 rounded-md bg-main-500 p-2 text-white transition"
      />
      <p class="mt-2 font-medium text-main-600">Sort + Filter</p>
      <p class="mt-4 text-xl font-bold text-gray-600">
        See the results you want.
      </p>
      <p class="mt-4 leading-snug text-gray-600">
        Full control over the results you see and their order. Don't settle for
        not seeing what you want.
      </p>
    </div>
    <div>
      <MapIcon
        aria-hidden="true"
        class="h-12 rounded-md bg-main-500 p-2 text-white transition"
      />
      <p class="mt-2 font-medium text-main-600">Made for Orienteering</p>
      <p class="mt-4 text-xl font-bold text-gray-600">It just works.</p>
      <p class="mt-4 leading-snug text-gray-600">
        Has all the options your league needs. Designed specifically for
        Orienteering, and to save volunteers time.
      </p>
    </div>
    <div>
      <DeviceMobileIcon
        aria-hidden="true"
        class="h-12 rounded-md bg-main-500 p-2 text-white transition"
      />
      <p class="mt-2 font-medium text-main-600">On the Go</p>
      <p class="mt-4 text-xl font-bold text-gray-600">Designed for Mobile.</p>
      <p class="mt-4 leading-snug text-gray-600">
        View all the information, on any device. See the results and manage
        leagues wherever you are.
      </p>
    </div>
    <NuxtLink
      to="/about"
      class="inline-block text-gray-600 hover:text-main-700 md:col-span-3 md:mt-4"
      no-prefetch
    >
      Read more about Munro, and why it was created â†’
    </NuxtLink>
  </section>
</template>
