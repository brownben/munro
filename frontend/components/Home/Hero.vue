<template>
  <header>
    <div
      class="mx-auto flex max-w-screen-lg flex-col items-center gap-10 px-8 py-6 sm:flex-row sm:py-10 md:py-14 lg:px-8 lg:py-20"
    >
      <div class="lg:-mt-4">
        <h1
          class="text-center text-4xl font-black tracking-tight text-gray-600 sm:text-left md:text-5xl"
        >
          League Results.
          <span
            class="block text-6xl text-main-600 sm:inline-block sm:text-7xl"
          >
            Sorted.
          </span>
        </h1>
        <NuxtLink
          to="/leagues"
          class="mt-6 hidden text-gray-500 hover:text-main-600 focus:text-main-600 lg:inline-block"
        >
          View All Leagues â†’
        </NuxtLink>
      </div>
      <HomeHeroImage class="w-full md:w-1/2" />
    </div>
    <HomeLogoCloud />
  </header>
</template>
