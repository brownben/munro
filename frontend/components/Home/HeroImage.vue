<template>
  <svg
    xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink"
    viewBox="0 0 411 259"
    fill="none"
    role="img"
    aria-label="Runner in front of hills with a league table in front of them"
  >
    <g fill="var(--hero-hills)">
      <path d="M129.5 199.797l-67.412-66.95L0 199.657" />
      <path
        d="M296.475 200H67.23l28.363-30.99L198.93 46.833l75.387 120.34L296.475 200z"
      />
      <path d="M410 199.797L325.617 99.44 246.33 199.657" />
    </g>
    <path
      fill="var(--hero-document)"
      d="M87.184 157.77l-14.94 10.097-17.64-7.005-23.627 5.458 31.1-33.475 25.097 24.925zM231.73 99.196l-5.005 8.863-11.222-11.85-12.296 11.85-6.95-13.467-11.226 13.467-3.208-8.617-7.48 8.617-13.365-8.617-9.45 3.437 47.402-56.046 32.8 52.362zm121.645 33.257l-13.86 17.095-11.225-13.47-17.105 14.008-12.396-16.74 26.828-33.907 27.76 33.013z"
      opacity=".05"
    />
    <path
      fill="var(--hero-clouds)"
      d="M66.466 71.674a19.1 19.1 0 00-8.13-4.883 18.98 18.98 0 00-9.458-.357 19.07 19.07 0 00-8.47 4.257A19.274 19.274 0 0035 78.521l35.13-1.64a19.264 19.264 0 00-3.666-5.205zm93.524-66a19.07 19.07 0 00-8.13-4.883 19.003 19.003 0 00-9.46-.357 19.07 19.07 0 00-8.469 4.257 19.179 19.179 0 00-5.4 7.83l35.132-1.64a19.24 19.24 0 00-3.666-5.205zm142.44 82.858a19.1 19.1 0 00-8.131-4.883 18.98 18.98 0 00-9.457-.357 19.07 19.07 0 00-8.47 4.257 19.283 19.283 0 00-5.408 7.83l35.13-1.64a19.24 19.24 0 00-3.665-5.205z"
    />
    <g fill="var(--hero-trees)">
      <path
        d="M320.313 247.578l17.486-25.65-17.45 28.4.036 2.886c-1.276.01-2.54-.01-3.792-.063l.867-37.1-.024-.286.032-.055.083-3.506L298.4 184.8l19.127 24.8.067.74.656-28.03-16.54-28.433 16.587 23.508-.167-58.075 1.097 45.77 15.074-18.592-15.024 22.535.33 25.072 13.87-24.47-13.82 28.156.183 13.942 20.172-34.086-20.1 38.95.4 31.002zm58.847.808l17.485-25.65-17.45 28.4.037 2.886c-1.277.01-2.54-.01-3.793-.063l.868-37.1-.024-.286.032-.055.083-3.506-19.15-27.424 19.126 24.8.068.74.655-28.03-16.54-28.432 16.586 23.508-.167-58.075 1.098 45.77 15.074-18.592-15.025 22.535.33 25.072 13.87-24.47-13.82 28.156.184 13.942 20.172-34.086-20.1 38.95.4 31.002z"
      />
      <path d="M411 253H63v1h348v-1z" />
    </g>
    <path
      d="M253.442 144.51s-2.365-17.113-6.308-12.446 0 11.668 0 11.668l3.154 3.89 3.154-2.334v-.777z"
      fill="#9f616a"
    />
    <use fill="var(--main-600)" xlink:href="#B" />
    <use fill="#000" opacity=".1" xlink:href="#B" />
    <path
      d="M307.854 149.176v6.222s.8 14.002-3.943 13.224-2.365-12.446-2.365-12.446l.788-7h5.52zm-2.366 45.114s7.097 10.9 9.463 11.668 14.982 7.778 16.56 14.78l11.04 14-5.52 7s-25.235-26.447-32.332-28.78-14.194-14-14.194-14l14.983-4.667zM249.5 176.4s-14.983 2.334-12.617 11.668 25.234 28.78 25.234 28.78l.8 4.668 8.674 1.555 3.943-3.9-.8-2.334s-9.463-8.556-10.25-11.667-9.463-15.557-13.406-16.335l5.52-3.1-7.097-9.335z"
      fill="#9f616a"
    />
    <path
      fill="var(--gray-800)"
      d="M340.185 234.74s-3.154-4.667 1.577-3.9 7.886 1.555 7.886 4.667-3.943 24.9-8.674 23.335-.8-8.556-.8-8.556-2.365-6.223-3.943-7-1.577-4.667 0-4.667 4.732-2.333 3.943-3.9zm-67.028-17.114s0-3.9 3.154 0 4.732 7 1.577 9.334-14.194 7.78-14.194 7.78-11.83 2.333-10.25-3.1c0 0 7.885-3.9 7.885-7s0-8.556 2.366-7 9.463 2.334 9.463 0zm14.983-58.338s11.04 1.555 10.25 17.112c0 0 14.195 14.002 10.252 18.67s-12.617 13.223-15.772 10.9-12.617-20.224-12.617-20.224-22.08 7.778-26.023 3.9-7.885-13.224-7.885-13.224L266.06 166.3l22.08-7z"
    />
    <g fill="#9f616a">
      <path
        d="M257.385 114.95c5.226 0 9.463-4.18 9.463-9.334s-4.237-9.334-9.463-9.334-9.463 4.18-9.463 9.334 4.237 9.334 9.463 9.334z"
      />
      <path
        d="M264.482 106.394s5.52 8.556 7.098 9.334-11.04 8.557-11.04 8.557 0-10.112-3.943-12.446l7.885-5.445z"
      />
    </g>
    <g fill="var(--main-600)">
      <path
        d="M265.27 114.95s12.617-3.89 14.983 2.334 7.886 17.9 7.097 25.67-2.366 6.223 0 8.556 3.943.778 3.154 3.89-1.577 3.112-.788 4.668-9.463 7-18.137 7.778l-8.675.778s0-14-1.577-17.113-4.73-9.334-4.73-11.667v-6.175c0-4.11 1.087-8.148 3.154-11.716 0 0 4.73-7 5.52-7z"
      />
      <path
        d="M274.734 118.062s29.177 2.333 32.33 10.112 4.732 23.335 1.577 23.335-7.097 1.556-7.885 0-3.943-17.112-4.732-17.112-24.446-2.334-24.446-2.334-10.25-12.446 3.155-14z"
      />
    </g>
    <path
      fill="var(--gray-800)"
      d="M257.005 103.302s-2.51-4.305-5.65-.566-8.245-3.132-7.46-4.598 3.552.066 8.296-1.464 13.882-4.646 15.413 5.52-2.115 9.1-2.115 9.1-1.67-9.008-3.18-6.42l-1.5 2.587-1.31.318-2.483-4.477z"
    />
    <path
      fill="var(--hero-document)"
      stroke="var(--hero-row)"
      d="M53.5 186.716V13.5h128v173.216h-128z"
    />
    <g fill="var(--hero-row)">
      <rect width="100.807" height="10.841" x="67.181" y="40.273" rx="1" />
      <rect width="100.807" height="10.841" x="67.181" y="61.955" rx="1" />
      <rect width="100.807" height="10.841" x="67.181" y="83.637" rx="1" />
      <rect width="100.807" height="10.841" x="67.181" y="105.319" rx="1" />
      <rect width="100.807" height="10.841" x="67.181" y="127.001" rx="1" />
      <rect width="100.807" height="10.841" x="67.181" y="148.683" rx="1" />
      <rect width="100.807" height="10.841" x="67.181" y="170.365" rx="1" />
    </g>
    <path
      fill="var(--main-500)"
      d="M67 26.5a2.5 2.5 0 012.5-2.5h39a2.5 2.5 0 110 5h-39a2.5 2.5 0 01-2.5-2.5z"
    />
    <defs>
      <path
        id="B"
        d="M262.905 149.954s-7.885-8.557-9.463-7-6.308 2.334-5.52 5.445 17.35 13.223 17.35 13.223l-2.366-11.667z"
      />
    </defs>
  </svg>
</template>
<style>
svg {
  --hero-document: white;
  --hero-hills: var(--gray-100);
  --hero-trees: var(--main-200);
  --hero-clouds: var(--main-300);
  --hero-row: var(--main-200);
}

@media (prefers-color-scheme: dark) {
  svg {
    --hero-document: var(--gray-900);
    --hero-hills: var(--gray-800);
    --hero-trees: var(--gray-500);
    --hero-clouds: var(--main-700);
    --hero-row: var(--gray-700);
  }
}
</style>
