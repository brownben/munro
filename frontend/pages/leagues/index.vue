<script setup lang="ts">
import type { League } from '~/api-types'

const { data: leagues } = await useData<League[]>('leagues/')

useTitle({
  title: 'Leagues',
  description: 'All the leagues with results on Munro',
})
</script>

<template>
  <div>
    <Heading title="Leagues" />
    <main
      class="mx-auto max-w-screen-lg gap-8 px-6 py-10 sm:py-12 md:py-14 lg:px-8"
    >
      <div
        class="col-span-2 -mx-4 -mb-4 -mt-8 flex flex-col divide-y dark:divide-gray-600"
      >
        <League
          v-for="league in leagues"
          :key="league.name"
          :league="league"
          class="px-4 py-10"
        />
      </div>
    </main>
  </div>
</template>
