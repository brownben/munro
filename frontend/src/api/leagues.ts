import { getData, postData, putData, deleteData } from './requests'

export const getLeague = (name: string): Promise<League | null> =>
  getData<League | null>({
    apiLocation: `/api/leagues/${name}`,
    customErrorMessage: 'Problem Fetching League Details',
  })

export const getLeagues = (): Promise<League[] | null> =>
  getData<League[]>({
    apiLocation: '/api/leagues',
    customErrorMessage: 'Problem Fetching Leagues',
  })

export const createLeague = (data: LeagueForm): Promise<ServerMessage | null> =>
  postData<ServerMessage>({
    apiLocation: `/api/leagues`,
    data,
    customErrorMessage: 'Problem Creating League',
    customSuccessMessage: `League \`${data.name}\` Created`,
    customErrorHandler: true,
  })

export const updateLeague = (data: LeagueForm): Promise<ServerMessage | null> =>
  putData<ServerMessage>({
    apiLocation: `/api/leagues/${data.name}`,
    data,
    customErrorMessage: 'Problem Updating League',
    customSuccessMessage: `League \`${data.name}\` Updated`,
    customErrorHandler: true,
  })

export const deleteLeague = (name: string): Promise<ServerMessage | null> =>
  deleteData<ServerMessage>({
    apiLocation: `/api/leagues/${name}`,
    customErrorMessage: 'Problem Deleting League',
    customErrorHandler: true,
  })
