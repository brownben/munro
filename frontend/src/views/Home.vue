<!--
  Home

  Home page of app, displaying summary of all the leagues
-->

<template>
  <div>
    <vue-headful
      :head="{
        meta: { name: 'robots', content: 'all' },
      }"
      title="Munro"
      description="Fast and Easy Results System for Orienteering Leagues by Munro, a simple way to calculate the results for orienteering leagues, with search and sort features"
      url="https://munro-leagues.herokuapp.com/"
    />
    <div
      id="header"
      class="flex-row-reverse items-center justify-center block p-3 pb-4 mt-6 text-center view md:flex"
    >
      <img
        src="@/assets/images/UnDrawOFlagPurple.svg"
        alt="Orienteering Flag Next to Trees and Hills"
        class="inline-block pb-2 h-80 md:pl-6 xl:pl-20"
      />
      <h2
        class="w-full pt-3 text-3xl text-center md:text-left font-heading md:text-4xl xl:text-5xl text-main md:py-0"
      >
        Fast and Easy Results for Orienteering Leagues
      </h2>
    </div>

    <div
      v-if="auth.user"
      class="w-full p-3 my-4 mt-6 text-center text-white actions bg-main-dark"
    >
      <h2 class="py-3 text-3xl font-heading">
        Hello {{ $auth.user.displayName || 'Admin' }}!
      </h2>
      <div class="pb-2">
        <router-link to="/create-league" class="button-white"
          >Create New League</router-link
        >
        <router-link to="/competitors" class="button-white"
          >Manage Competitors</router-link
        >
        <router-link to="/upload" class="button-white"
          >Upload Results</router-link
        >
      </div>
    </div>

    <div class="flex flex-row flex-wrap view">
      <div class="w-full my-4 md:w-1/2 md:px-3">
        <div class="w-full h-full card">
          <img
            src="@/assets/images/SprintelopeLogo.png"
            alt="The Logo of Sprintelope"
            class="h-40"
          />
          <h1 class="my-2 text-2xl font-heading">(Not) Sprintelope 2020</h1>
          <p>
            Socially Distant Sprint/ Urban Orienteering Events in Edinburgh and
            the Lothians
          </p>
          <p>
            More information can be found on their
            <a
              href="https://en-gb.facebook.com/sprintelope/"
              target="_blank"
              rel="noopener noreferrer"
              class="link"
              >website</a
            >
          </p>
          <router-link class="button" to="/leagues/(Not) Sprintelope 2020"
            >View League</router-link
          >
        </div>
      </div>

      <div class="w-full h-full my-4 md:w-1/2 md:px-3">
        <div class="card">
          <img
            src="@/assets/images/SprintelopeLogo.png"
            alt="The Logo of Sprintelope"
            class="h-40"
          />
          <h1 class="my-2 text-2xl font-heading">Sprintelope 2019</h1>
          <p>
            Wednesday Evening Sprint Orienteering Events in Edinburgh and the
            Lothians
          </p>
          <p>
            More information can be found on their
            <a
              href="https://en-gb.facebook.com/sprintelope/"
              target="_blank"
              rel="noopener noreferrer"
              class="link"
              >website</a
            >
          </p>
          <router-link class="button" to="/leagues/Sprintelope 2019"
            >View League</router-link
          >
        </div>
      </div>

      <div class="w-full my-4 md:w-1/2 md:px-3">
        <div class="w-full h-full card">
          <h1 class="my-2 text-2xl font-heading">Fight with the Night</h1>
          <p>
            Thursday Night Orienteering Events Around Edinburgh Organised By
            Edinburgh Uni Orienteering Club
          </p>
          <p>
            More information can be found on their
            <a
              href="https://euoc.wordpress.com/"
              target="_blank"
              rel="noopener noreferrer"
              class="link"
              >website</a
            >
          </p>
          <router-link class="button" to="/leagues/Fight with the Night 19-20"
            >View League</router-link
          >
        </div>
      </div>
      <div class="w-full my-4 md:w-1/2 md:px-3">
        <div class="w-full h-full card">
          <h1 class="my-2 text-2xl font-heading">Northern Night Cup</h1>
          <p>
            A Series of Night Orienteering Events in the North of Scotland
          </p>
          <p>
            More information can be found on their
            <a
              href="https://www.orienteering.scot/northern-night-cup/"
              target="_blank"
              rel="noopener noreferrer"
              class="link"
              >website</a
            >
          </p>
          <router-link class="button" to="/leagues/Northern Night Cup 19-20"
            >View League</router-link
          >
        </div>
      </div>
    </div>
    <div class="w-full p-3 text-center text-white bg-main view contours">
      <img
        src="@/assets/images/MunroLogoWhite.png"
        alt="Munro Logo"
        class="inline-block h-24 mt-4 mb-2"
      />
      <h1 class="my-2 text-2xl font-heading">About</h1>
      <div class="mb-3 text">
        <p>
          Munro is a sports league results calculator, created originally by Ben
          Brown for his local Orienteering League as part of his Advanced Higher
          Computing Project. Munro is designed to be as easy to use as possible
          providing all the features and information you need, whilst making it
          fast and simple to view the results; with multiple options to sort and
          filter them. It supports multiple file formats for upload and several
          different scoring systems, making the process as quick and stress free
          as possible for event organisers.
        </p>
        <p>
          Munro is Open Source and the code is avaliable on
          <a class="link" href="https://github.com/brownben/munro">Github</a>
        </p>
        <p>
          Munro is still under development so any feedback or comments would be
          welcome, please send them to munro.leagues@gmail.com
        </p>
        <p>
          For any help or enquires please email munro.leagues@gmail.com
        </p>
      </div>
    </div>
    <div class="flex flex-row flex-wrap view">
      <div class="w-full my-4 md:w-1/2 md:px-3">
        <div class="w-full h-full card">
          <h1 class="my-2 text-2xl font-heading">Upload Instructions</h1>
          <p>
            Instructions for Event Organisers on how to Upload Results to Munro
          </p>
          <router-link to="/upload-instructions" class="button"
            >View Info</router-link
          >
        </div>
      </div>
      <div class="w-full my-4 md:w-1/2 md:px-3">
        <div class="w-full h-full card">
          <h1 class="my-2 text-2xl font-heading">Developers</h1>
          <p>
            Information for developers about accessing the API and embedding the
            site in IFrames
          </p>
          <router-link to="/developers" class="button">View Info</router-link>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  data: function () {
    return {
      auth: this.$auth,
    }
  },
}
</script>
<style lang="postcss">
.h-80 {
  height: 18rem;
}

.text p {
  @apply my-1;
}

#header img {
  max-width: 100%;
}

@media (min-width: 768px) {
  #header img {
    height: auto;
    max-height: 20rem;
  }
}
.contours {
  background-image: url('../assets/images/topography.svg');
}
</style>
