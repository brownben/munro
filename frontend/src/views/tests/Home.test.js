/*
  Home View Unit Tests
*/

import axios from 'axios'
import { mount } from '@vue/test-utils'
import Home from '@/views/Home'
import { sampleSingleLeague, sampleDataThreeLeagues } from '@/tests/test data/leagues'

jest.mock('axios')

beforeEach(() => {
  jest.clearAllMocks()
  axios.get.mockResolvedValue({ data: [] })
})

test('Is a Vue Instance', () => {
  const wrapper = mount(Home, {
    mocks: {
      $auth: { user: false },
    },
    stubs: ['router-link', 'vue-headful'],
  })
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly - No Leagues', () => {
  const wrapper = mount(Home, {
    mocks: {
      $auth: { user: false },
    },
    stubs: ['router-link', 'vue-headful'],
  })
  expect(wrapper.element).toMatchSnapshot()
})

test('Renders Correctly - One League', async () => {
  axios.get.mockResolvedValue({ data: sampleSingleLeague })
  const wrapper = mount(Home, {
    mocks: {
      $auth: { user: false },
    },
    stubs: ['router-link', 'vue-headful'],
  })
  await wrapper.vm.getLeagues()
  expect(wrapper.element).toMatchSnapshot()
})

test('Renders Correctly - Multiple Leagues', async () => {
  axios.get.mockResolvedValue({ data: sampleDataThreeLeagues })
  const wrapper = mount(Home, {
    mocks: {
      $auth: { user: false },
    },
    stubs: ['router-link', 'vue-headful'],
  })
  await wrapper.vm.getLeagues()
  expect(wrapper.element).toMatchSnapshot()
})

test('Shows Admin Buttons When Logged In', () => {
  const wrapper = mount(Home, {
    mocks: {
      $auth: { user: true },
    },
    stubs: ['router-link', 'vue-headful'],
  })
  expect(wrapper.find('.actions').isVisible()).toBeTruthy()
  expect(wrapper.find('.actions').findAll('router-link-stub').length).toBe(3)
  expect(wrapper.find('.actions').findAll('router-link-stub').at(0).text()).toBe('Create New League')
  expect(wrapper.find('.actions').findAll('router-link-stub').at(1).text()).toBe('Upload Results')
})

test('Get Leagues - Request Called Correctly', async () => {
  const mockAddMessageFunction = jest.fn()
  const wrapper = mount(Home, {
    mocks: {
      $auth: { user: false },
      $messages: { addMessage: mockAddMessageFunction },
    },
    stubs: ['router-link', 'vue-headful'],
  })
  jest.resetAllMocks()
  axios.get.mockResolvedValue({ data: sampleSingleLeague })
  await wrapper.vm.getLeagues()
  expect(axios.get).toHaveBeenCalledTimes(1)
  expect(axios.get).toHaveBeenLastCalledWith('/api/leagues')
})

test('Get Leagues - Processes Response Correctly', async () => {
  const mockAddMessageFunction = jest.fn()
  const wrapper = mount(Home, {
    mocks: {
      $auth: { user: false },
      $messages: { addMessage: mockAddMessageFunction },
    },
    stubs: ['router-link', 'vue-headful'],
  })
  axios.get.mockResolvedValue({ data: [{ league: '1' }, { league: '2' }] })
  await wrapper.vm.getLeagues()
  expect(wrapper.vm.leagues).toEqual([{ league: '1' }, { league: '2' }])

  axios.get.mockResolvedValue({ data: [] })
  await wrapper.vm.getLeagues()
  expect(wrapper.vm.leagues).toEqual([])

  axios.get.mockResolvedValue({ data: [{ league: 'a', property: 3 }, { league: '2', another: 4 }] })
  await wrapper.vm.getLeagues()
  expect(wrapper.vm.leagues).toEqual([{ league: 'a', property: 3 }, { league: '2', another: 4 }])
})

test('Get Leagues - Shows Message on Error', async () => {
  const mockAddMessageFunction = jest.fn()
  const wrapper = mount(Home, {
    mocks: {
      $auth: { user: false },
      $messages: { addMessage: mockAddMessageFunction },
    },
    stubs: ['router-link', 'vue-headful'],
  })
  axios.get.mockRejectedValue()
  await wrapper.vm.getLeagues()
  expect(wrapper.vm.leagues).toEqual([])
  expect(mockAddMessageFunction).toHaveBeenCalledTimes(1)
  expect(mockAddMessageFunction).toHaveBeenLastCalledWith('Problem Fetching League Details')
})
