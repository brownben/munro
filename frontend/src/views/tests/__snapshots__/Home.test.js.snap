// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly - Multiple Leagues 1`] = `
<div>
  <vue-headful-stub
    description="Fast and Easy Results for Orienteering Leagues. A simple way to calculate the results for orienteering leagues, with search and sort features"
    title="Munro - League Results for Orienteering"
  />
   
  <div
    id="header"
  >
    <img
      alt="Runner Crossing Finsh Line"
      src="@/assets/images/UnDrawRunningWhite.svg"
    />
     
    <h2>
      Fast and Easy Results for Orienteering Leagues
    </h2>
  </div>
   
  <!---->
   
  <div
    id="leagues"
  >
    <div
      class="league"
    >
      <img
        alt="The Logo of Sprintelope"
        height="150px"
        src="@/assets/images/SprintelopeLogo.png"
      />
       
      <h1>
        Sprintelope 2019
      </h1>
       
      <p>
        Wednesday Evening Sprint Orienteering Events in Edinburgh and the Lothians
      </p>
       
      <p>
        
        More information can be found at
        
        <a
          href="https://en-gb.facebook.com/sprintelope/"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://en-gb.facebook.com/sprintelope/
        </a>
      </p>
       
      <router-link-stub
        class="button"
        to="/leagues/Sprintelope 2019"
      >
        View League
      </router-link-stub>
    </div>
     
    <div
      class="league"
    >
      <img
        alt="The Logo of Sprintelope"
        height="150px"
        src="@/assets/images/SprintelopeLogo.png"
      />
       
      <h1>
        Sprintelope 2018
      </h1>
       
      <p>
        Wednesday Evening Sprint Orienteering Events in Edinburgh and the Lothians
      </p>
       
      <p>
        
        More information can be found at
        
        <a
          href="https://en-gb.facebook.com/sprintelope/"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://en-gb.facebook.com/sprintelope/
        </a>
      </p>
       
      <router-link-stub
        class="button"
        to="/leagues/Sprintelope 2018"
      >
        View League
      </router-link-stub>
    </div>
     
    <div
      class="card"
      id="about"
    >
      <img
        alt="Munro Logo"
        src="@/assets/images/MunroLogo.png"
      />
       
      <h1>
        About
      </h1>
       
      <div
        class="text"
      >
        <p>
          Munro was created by Ben Brown for his Advanced Higher Computing Project, as a calculator of the orienteering league standings. It is designed to be fast and easy to use, offering sorting and filtering on the results. It also has easy upload supporting various upload formats to make the upload as quick and easy as possible for event organisers.
        </p>
         
        <p>
          Munro is still under development so any feedback or comments would be welcome, please send them to munro.leagues(a)gmail.com
        </p>
         
        <p>
          For any help or enquires please email munro.leagues(a)gmail.com
        </p>
      </div>
    </div>
     
    <div
      class="card middle"
    >
      <h1>
        Developers
      </h1>
       
      <p>
        Information for developers about accessing the API and embedding the site in IFrames
      </p>
       
      <router-link-stub
        class="button"
        to="/developers"
      >
        View Info
      </router-link-stub>
    </div>
  </div>
   
  <p
    class="credits"
  >
    
    © Ben Brown 2019
    
    <router-link-stub
      to="/login"
    >
      Admin Login
    </router-link-stub>
  </p>
</div>
`;

exports[`Renders Correctly - No Leagues 1`] = `
<div>
  <vue-headful-stub
    description="Fast and Easy Results for Orienteering Leagues. A simple way to calculate the results for orienteering leagues, with search and sort features"
    title="Munro - League Results for Orienteering"
  />
   
  <div
    id="header"
  >
    <img
      alt="Runner Crossing Finsh Line"
      src="@/assets/images/UnDrawRunningWhite.svg"
    />
     
    <h2>
      Fast and Easy Results for Orienteering Leagues
    </h2>
  </div>
   
  <!---->
   
  <div
    id="leagues"
  >
    <div
      class="league"
    >
      <img
        alt="The Logo of Sprintelope"
        height="150px"
        src="@/assets/images/SprintelopeLogo.png"
      />
       
      <h1>
        Sprintelope 2019
      </h1>
       
      <p>
        Wednesday Evening Sprint Orienteering Events in Edinburgh and the Lothians
      </p>
       
      <p>
        
        More information can be found at
        
        <a
          href="https://en-gb.facebook.com/sprintelope/"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://en-gb.facebook.com/sprintelope/
        </a>
      </p>
       
      <router-link-stub
        class="button"
        to="/leagues/Sprintelope 2019"
      >
        View League
      </router-link-stub>
    </div>
     
    <div
      class="league"
    >
      <img
        alt="The Logo of Sprintelope"
        height="150px"
        src="@/assets/images/SprintelopeLogo.png"
      />
       
      <h1>
        Sprintelope 2018
      </h1>
       
      <p>
        Wednesday Evening Sprint Orienteering Events in Edinburgh and the Lothians
      </p>
       
      <p>
        
        More information can be found at
        
        <a
          href="https://en-gb.facebook.com/sprintelope/"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://en-gb.facebook.com/sprintelope/
        </a>
      </p>
       
      <router-link-stub
        class="button"
        to="/leagues/Sprintelope 2018"
      >
        View League
      </router-link-stub>
    </div>
     
    <div
      class="card"
      id="about"
    >
      <img
        alt="Munro Logo"
        src="@/assets/images/MunroLogo.png"
      />
       
      <h1>
        About
      </h1>
       
      <div
        class="text"
      >
        <p>
          Munro was created by Ben Brown for his Advanced Higher Computing Project, as a calculator of the orienteering league standings. It is designed to be fast and easy to use, offering sorting and filtering on the results. It also has easy upload supporting various upload formats to make the upload as quick and easy as possible for event organisers.
        </p>
         
        <p>
          Munro is still under development so any feedback or comments would be welcome, please send them to munro.leagues(a)gmail.com
        </p>
         
        <p>
          For any help or enquires please email munro.leagues(a)gmail.com
        </p>
      </div>
    </div>
     
    <div
      class="card middle"
    >
      <h1>
        Developers
      </h1>
       
      <p>
        Information for developers about accessing the API and embedding the site in IFrames
      </p>
       
      <router-link-stub
        class="button"
        to="/developers"
      >
        View Info
      </router-link-stub>
    </div>
  </div>
   
  <p
    class="credits"
  >
    
    © Ben Brown 2019
    
    <router-link-stub
      to="/login"
    >
      Admin Login
    </router-link-stub>
  </p>
</div>
`;

exports[`Renders Correctly - One League 1`] = `
<div>
  <vue-headful-stub
    description="Fast and Easy Results for Orienteering Leagues. A simple way to calculate the results for orienteering leagues, with search and sort features"
    title="Munro - League Results for Orienteering"
  />
   
  <div
    id="header"
  >
    <img
      alt="Runner Crossing Finsh Line"
      src="@/assets/images/UnDrawRunningWhite.svg"
    />
     
    <h2>
      Fast and Easy Results for Orienteering Leagues
    </h2>
  </div>
   
  <!---->
   
  <div
    id="leagues"
  >
    <div
      class="league"
    >
      <img
        alt="The Logo of Sprintelope"
        height="150px"
        src="@/assets/images/SprintelopeLogo.png"
      />
       
      <h1>
        Sprintelope 2019
      </h1>
       
      <p>
        Wednesday Evening Sprint Orienteering Events in Edinburgh and the Lothians
      </p>
       
      <p>
        
        More information can be found at
        
        <a
          href="https://en-gb.facebook.com/sprintelope/"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://en-gb.facebook.com/sprintelope/
        </a>
      </p>
       
      <router-link-stub
        class="button"
        to="/leagues/Sprintelope 2019"
      >
        View League
      </router-link-stub>
    </div>
     
    <div
      class="league"
    >
      <img
        alt="The Logo of Sprintelope"
        height="150px"
        src="@/assets/images/SprintelopeLogo.png"
      />
       
      <h1>
        Sprintelope 2018
      </h1>
       
      <p>
        Wednesday Evening Sprint Orienteering Events in Edinburgh and the Lothians
      </p>
       
      <p>
        
        More information can be found at
        
        <a
          href="https://en-gb.facebook.com/sprintelope/"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://en-gb.facebook.com/sprintelope/
        </a>
      </p>
       
      <router-link-stub
        class="button"
        to="/leagues/Sprintelope 2018"
      >
        View League
      </router-link-stub>
    </div>
     
    <div
      class="card"
      id="about"
    >
      <img
        alt="Munro Logo"
        src="@/assets/images/MunroLogo.png"
      />
       
      <h1>
        About
      </h1>
       
      <div
        class="text"
      >
        <p>
          Munro was created by Ben Brown for his Advanced Higher Computing Project, as a calculator of the orienteering league standings. It is designed to be fast and easy to use, offering sorting and filtering on the results. It also has easy upload supporting various upload formats to make the upload as quick and easy as possible for event organisers.
        </p>
         
        <p>
          Munro is still under development so any feedback or comments would be welcome, please send them to munro.leagues(a)gmail.com
        </p>
         
        <p>
          For any help or enquires please email munro.leagues(a)gmail.com
        </p>
      </div>
    </div>
     
    <div
      class="card middle"
    >
      <h1>
        Developers
      </h1>
       
      <p>
        Information for developers about accessing the API and embedding the site in IFrames
      </p>
       
      <router-link-stub
        class="button"
        to="/developers"
      >
        View Info
      </router-link-stub>
    </div>
  </div>
   
  <p
    class="credits"
  >
    
    © Ben Brown 2019
    
    <router-link-stub
      to="/login"
    >
      Admin Login
    </router-link-stub>
  </p>
</div>
`;
